File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
         var map = new HashMap<Integer, Integer>();

         for(int i = 0; i < nums.length; i++){
            var index = map.getOrDefault(target - nums[i], -1);
            if (index >= 0){
                return new int[]  {index, i};
            }
            map.put(nums[i], i);
         }

         return null;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        lo, hi = 0, len(nums) - 1

        while lo < hi:
            mid = (lo + hi) // 2
            if nums[mid] > nums[hi]:
                lo = mid + 1        # mid в левом куске, минимум правее
            else:
                hi = mid            # mid в правом куске, может быть ответом

        return nums[lo]